### PR TITLE
feat(StyleGuide): update layout templates, removing pure, adding flex

### PR DESCRIPTION
**Pull request by TheNando**
- [ ] Dev LGTM
- [ ] Dev LGTM
- [ ] Design LGTM
- [ ] QE LGTM

Addresses #1004

@Peer Reviewers, use https://github.com/rackerlabs/encore-ui/pull/1031/files?w=1 to ignore whitespace during review.

### CODE REVIEW

**Comment by CITguy** [comment]
LGTM given #1028 is merged before or with this PR

**Comment by Droogans** [comment]
@TheNando did you mean to make the div for the page actions so tall?

https://github.com/rackerlabs/encore-ui-screenshots/pull/16/files#diff-c8888eead74b7a4afbe904bdbb7b3b02

**Comment by CITguy** [comment]
@Droogans, I'm wondering if there's a glitch with the visual regression around that. Locally, the page actions look like this (sans blue border):

screen shot 2015-06-10 at 2 04 51 pm

**Comment by CITguy** [comment]
Also, the screen resolution plays into the height.

**568x320**
layout 1 encoreui

**Comment by glynnis** [comment]
It looks like the page actions got moved. They should be adjacent to the page title, not the metadata section.

For comparison:

### Billing (correct page actions position)
screen shot 2015-06-11 at 9 34 27 am

### Cloud (correct page actions position)
screen shot 2015-06-11 at 9 34 09 am

### Layout 1 (page actions position too low)
screen shot 2015-06-11 at 9 32 31 am

**Comment by TheNando** [comment]
That somebody is me. I'll fix it up.

**Comment by CITguy** [comment]
I think it may just be a matter of rearranging items. In rough pseudo code, you might consider this layout
```
#row
  #col
    Title & Subtitle
  #col
    Actions
#row
  #col
    page content
```
This way, the title and actions will be in the same row (as expected with the current layout), and the content will be in rows below that.  The issue that will arise is when you have lots of actions that push the top row height further down the page, but this is already something that we run into today with the existing layout. (see image below - Server Details)

*STAGING: Server Details*
screen shot 2015-06-17 at 9 48 23 am
* blue : would translate to a `layout="row"`
* red : would translate as a `flex` attribute

**Comment by TheNando** [comment]
The problem is the entry point into the page. rxPage's renders to this

```
<rx-page>
   <header></headers>
   <div class="page-body>
      <div class="rx-notifications></div>
      <div class="page-titles"></div>
      <div class="page-content">
         {
            rx-page content injected here, including:
            <div class="page-actions>
         }
      </div>
      ...
   </div>
</rx-page>
```

Roughly, with the layout module, it looks like this, with the page actions not aligned to the top:
image

The red is the first entry point and shows how the layout is separated out into collections of rows and columns.

I think this layout below would work really well, but we'd need to address the window bar title and how to programmatically place the title H1:

image

File: demo/styleguide/layout-3.html [hunk 1]
```diff
@@ -1,86 +1,88 @@
 <rx-page title="'Layout 3'" subtitle="'Sample Create Page'">
 
-<form class="rx-form clear">
```
Comment (CITguy) [inline demo/styleguide/layout-3.html]: Considering that this is purely cosmetic for code style, (and considering that my rxForm stuff heavily modifies this file), I would feel better if this change was removed from the PR.

**Comment by parlarjb** [comment]
Current status on this?

**Comment by TheNando** [comment]
@parlarjb: I'd like to discuss the issue with the team in more detail and see if we can come up with a solution.

**Comment by Droogans** [comment]
@TheNando Did this ever get resolved?

**Comment by TheNando** [comment]
I'm closing this without it being resolved. I really think the best solution is to add a new rxFlexPage directive that will avoid all these cases.

**Pull request closed without merging**